geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java

```diff
@@ -693,15 +693,19 @@ public boolean put(Object object) throws InterruptedException, CacheException {
 
     boolean isDREvent = isDREvent(sender.getCache(), value);
 
-    Region region = value.getRegion();
-    String regionPath = null;
-    if (isDREvent) {
-      regionPath = region.getFullPath();
-    } else {
+    String regionPath = value.getRegionPath();
+    if (!isDREvent) {
+      Region region = sender.getCache().getRegion(regionPath);
+      if (region == null) {
+        if (isDebugEnabled) {
+          logger.debug("The PR " + regionPath + " has not finished initializing.");
+        }
+        region = value.getRegion();
```

Comment:
Should not use value.getRegion() here as it will still cause the hang.

Reply:
After I use getRegion(path, true), I don't need to call value.getRegion(). 

Otherwise, value.getRegion() is still necessary. When code arrived here, it's normal enqueue (not syncWith enqueue), the event.getRegion() will never be null.